internal/cloud/realm/auth_test.go

```diff
@@ -49,6 +49,41 @@ func TestRealmAuthProfile(t *testing.T) {
 	})
 }
 
+func TestRealmAuthRefresh(t *testing.T) {
+	u.SkipUnlessRealmServerRunning(t)
+
+	t.Run("Does not refresh auth if request does not return invalid session code", func(t *testing.T) {
+		client := realm.NewClient(u.RealmServerURL())
+
+		session, err := client.Authenticate(u.CloudUsername(), u.CloudAPIKey())
+		assert.Equal(t, nil, err)
+
+		// invalidate the access token
+		session.AccessToken = session.RefreshToken
+
+		client = realm.NewAuthClient(u.RealmServerURL(), session)
+		_, err = client.AuthProfile()
+		serverError := err.(realm.ServerError)
+		assert.Equal(t, "invalid session: valid Issuer required", serverError.Message)
```

Comment:
[meganit] any chance we could get this assert to check Equal on the `ServerError` struct itself here?